component/background/RoomService.tsx

```diff
@@ -50,19 +54,17 @@ class RoomService {
     });
 
     // set up configRef
-    this.userListRef = this.rootRef.child("user");
-    this.myConfRef = this.userListRef.child(this.id);
-    this.myConfRef.set(true);
-
+    this.rootRef.child(USER).child(this.id).set(true);
+    this.myself = allUsers.child(this.id);
     this.active = true;
   }
 
   public close(): void {
     if (this.active) {
       this.messageRef.off();
-      this.myConfRef.remove()
+      this.rootRef.child(USER).child(this.id).remove()
         .then(() => {
-          this.userListRef.once("value", (data) => {
+          this.rootRef.child(USER).once("value", (data) => {
```

Comment:
Since `this.rootRef.child(USER)` is being used twice, I think it would be a good idea to store it a variable. Furthermore, rather than chaining calls, I think it is more descriptive if you broke it up with proper variable names.